wallsAndGates.js

```diff
@@ -0,0 +1,122 @@
+/*
+You are given an m x n grid rooms initialized with these three possible values.
+
+- -1 A wall or an obstacle.
+- 0 A gate.
+- INF Infinity means an empty room. We use the value 2^31 - 1 = 2147483647 to represent INF as you may assume that the distance to a gate is less than 2147483647.
+
+Fill each empty room with the distance to its nearest gate. If it is impossible to reach a gate, it should be filled with INF.
+*/
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [0, 1],
+    [1, 0],
+    [0, -1],
+    [-1, 0],
+];
+
+const GATE = 0;
+const WALL = -1;
+
+const isInBound = (matrix, row, col) => {
+    const rowBoundLength = matrix.length;
+    const colBoundLength = matrix[0].length;
+
+    const isInRowBound = row >= 0 && row < rowBoundLength;
+    const isInColBound = col >= 0 && col < colBoundLength;
+
+    return isInRowBound && isInColBound;
+};
+
+const getNeighbors = (matrix, row, col) => {
+    const neighbors = [];
+
+    for (const direction of directions) {
+        const [rowDir, colDir] = direction;
+
+        const newRow = row + rowDir;
+        const newCol = col + colDir;
+
+        if (!isInBound(matrix, newRow, newCol) || matrix[newRow][newCol] === WALL) continue;
+
+        neighbors.push({ row: newRow, col: newCol });
+    }
+
+    return neighbors;
+};
+
+const getLocationString = (gateLocation) => {
+    const { row, col } = gateLocation;
+
+    return `${row}, ${col}`;
+};
+
+const getLocations = (matrix) => {
```

Comment:
`getLocations` is not a clear name, I have no idea what this actually does without reading the code. Maybe `getGateLocations` would be better.